src/canvas/Geometry/BezierCurve.tsx

```diff
@@ -5,31 +5,40 @@ export class BezierCurve extends Curve {
     // Precondition: split is a point on the curve.
     split = (point: Point): BezierCurve[] => {
         const curves:  BezierCurve[] = [];
+        const t = this._findT(point);
 
-        const NUMPOINTS = 100;
-        const points = this.computePoints();
-        const startPoint = points[0];
-        const endPoint = points[points.length - 1];
-
-        let index = -1;
-        points.forEach((pt, i) => { 
-            if (pt.isWithinRadius(point, 10)) {
-                index = i;
-                return true;
-            }
-        });
-        console.log("BezierCurveIndex split index", index);
-
-        const t: number = index / NUMPOINTS;
-        let controlPointX = (1 - t) * startPoint.x + t * this.control.x;
-        let controlPointY = (1 - t) * startPoint.y + t * this.control.y;
-        curves.push(new BezierCurve(startPoint, point, new Point(controlPointX, controlPointY)));
-
-        controlPointX = (1 - t) * this.control.x + t * this.end.x;
-        controlPointY = (1 - t) * this.control.y + t * this.end.y;
-        curves.push(new BezierCurve(point, endPoint, new Point(controlPointX, controlPointY)));
+        let controlPointX = this.lerp(this.start.x, this.control.x, t);
+        let controlPointY = this.lerp(this.start.y, this.control.y, t);
+        curves.push(new BezierCurve(this.start, point, new Point(controlPointX, controlPointY)));
+
+        controlPointX = this.lerp(this.control.x, this.end.x, t);
+        controlPointY = this.lerp(this.control.y, this.end.y, t);
+        curves.push(new BezierCurve(point, this.end, new Point(controlPointX, controlPointY)));
 
         return curves;
+    }; 
+
+    private _solveQuadratic = (a: number, b: number, c: number): number[] => {
+        const result: number[] = [];
+        const d = b * b - 4 * a * c; // d is the discriminant
+        if (d === 0) {
+            result.push((-1 * b) / (2 * a));
+        } else if (d > 0) {
+            result.push(((-1 * b) + Math.sqrt(d)) / (2 * a));
+            result.push(((-1 * b) - Math.sqrt(d)) / (2 * a));
+        }
+        return result;
+    };
+
+    private _findT = (point: Point): number => {
+        const solx = this._solveQuadratic(this.start.x - 2 * this.control.x + this.end.x, 2 * (this.control.x - this.start.x), this.start.x - point.x);
```

Comment:
Why subtract 2? It seems arbitrary to me, but I am sure its not :)

Reply:
Because of the priority of operations, it's this.start.x - ( 2 * this.control.x) + this.end.x.
do you think I should add parentheses for clarity?

Reply:
Oh of course, I did not read it closely enough. Thanks!